src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -295,6 +325,98 @@ void ScheduleTask(DateTimeOffset utcNow)
             }
         }
 
+        private readonly List<WeakReference<Stats>?>? _allStats;
+        private readonly Stats? _accumulatedStats;
+        private ThreadLocal<Stats>? _stats;
+
+        private (long, long) Sum()
+        {
+            long hits = 0, misses = 0;
+            lock (_allStats!)
+            {
+                hits += _accumulatedStats!.Hits;
```

Comment:
You could not set the hits and misses to 0 to start with, and just set them here.

```C#
            long hits, misses;
            lock (_allStats!)
            {
                hits = _accumulatedStats!.Hits;
                misses = _accumulatedStats!.Misses;

```